### PR TITLE
[bitnami/mongodb-sharded] Add MongoDB Sharded chart

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@vmware.com>

**Description of the change**

Create a MongoDB Sharded chart. It is a MongoDB configuration meant for large datasets.

Supports:
   - HA thanks to the sharded and replica set configuration
   - Custom Init Scripts
   - Custom Sidecars and Init Containers
   - Custom configuration
   - Custom ports
   - Prometheus metrics
 
 
**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/mongodb-sharded/values.yaml [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+  ##
+  extraVolumes: []
+  ## Array to add extra mounts (normally used with extraVolumes)
+  ##
+  extraVolumeMounts: []
+  ## Use an alternate scheduler, e.g. "stork".
+  ## ref: https://kubernetes.io/docs/tasks/administer-cluster/configure-multiple-schedulers/
+  ##
+  schedulerName:
+
```
Comment (carrodher) [inline bitnami/mongodb-sharded/values.yaml]: Extra line

File: bitnami/mongodb-sharded/values.yaml [hunk 2]
```diff
@@ -0,0 +1,744 @@
...
+  accessModes:
+    - ReadWriteOnce
+  ## PersistentVolumeClaim size
+  ##
+  size: 8Gi
+  ## Additional volume annotations
+  ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+  ##
+  annotations: {}
+
```
Comment (carrodher) [inline bitnami/mongodb-sharded/values.yaml]: Another one

File: bitnami/mongodb-sharded/README.md [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+### Sharding
+
+This chart deploys a sharded cluster by default. Some characteristics of this chart are:
+
+- It allows HA by enabling replication on the shards and the config servers. The mongos instances can be scaled horizontally as well.
+- The number of secondary and arbiter nodes can be scaled out independently.
+
+### Initialize a fresh instance
+
+The [Bitnami MongoDB](https://github.com/bitnami/bitnami-docker-mongodb) image allows you to use your custom scripts to initialize a fresh instance. You can create a custom config map and give it via `initScriptsCM`(check options for more details).
```
Comment (tompizmor) [inline bitnami/mongodb-sharded/README.md]: should you refer to the mongodb-sharded image? (In the github link)

File: bitnami/mongodb-sharded/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+  {{- else -}}
+    {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+  {{- end -}}
+{{- end -}}
+
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "mongodb-sharded.imagePullSecrets" -}}
```
Comment (tompizmor) [inline bitnami/mongodb-sharded/templates/_helpers.tpl]: There is a simpler way of defining these imagePullSecrets helper templates. Check minio for example and the use of the coalesce function.

https://github.com/bitnami/charts/blob/master/bitnami/minio/templates/_helpers.tpl#L54-L65

File: bitnami/mongodb-sharded/templates/mongos/mongos-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+              {{- include "mongodb-sharded.tplValue" ( dict "value" $.Values.mongos.extraVolumeMounts "context" $ ) | nindent 12 }}
+            {{- end }}
+          resources: {{- toYaml .Values.mongos.resources | nindent 12 }}
+        {{- if .Values.metrics.enabled }}
+        - name: metrics
+          image: {{ include "mongodb-sharded.metrics.image" . }}
+          imagePullPolicy: {{ .Values.metrics.image.pullPolicy | quote }}
+          {{- if .Values.securityContext.enabled }}
+          securityContext:
+            runAsNonRoot: true
```
Comment (tompizmor) [inline bitnami/mongodb-sharded/templates/mongos/mongos-deployment.yaml]: If we ever want to run this as root we would need to edit the template. Can you parametrize this?

File: bitnami/mongodb-sharded/values-production.yaml [hunk 6]
```diff
@@ -0,0 +1,753 @@
...
+#     - myRegistryKeySecretName
+#   storageClass: myStorageClass
+
+## Bitnami MongoDB Sharded image version
+## ref: https://hub.docker.com/r/bitnami/mongodb-sharded/tags/
+##
+image:
+  registry: docker.io
+  repository: bitnami/mongodb-sharded
+  tag: "4.0"
```
Comment (tompizmor) [inline bitnami/mongodb-sharded/values-production.yaml]: use full tag

Comment (javsalgar) [reply bitnami/mongodb-sharded/values-production.yaml]: When I release the container

**Pull request merged**